core/include/moveit/task_constructor/storage.h

```diff
@@ -82,8 +82,8 @@ class InterfaceState
 	enum Status
 	{
 		ENABLED,  // state is actively considered during planning
+		FAILED,  // state that failed, causing the whole partial solution to be disabled
 		DISABLED,  // state is disabled because a required connected state failed
```

Comment:
I had to reorder this because of [early stopping in `hasPendingOpposites`](https://github.com/ros-planning/moveit_task_constructor/pull/287/files#diff-b284e09d92fea0042ecfec277496e7e6fdccc48459e7a2d008c8be34bb33ad57R749).
It's a matter of perspective which one is "more severe" here. `FAILED` might be retried in `Connecting` stages, but `DISABLED` will never be looked at unless it becomes active again.

Reply:
https://github.com/ros-planning/moveit_task_constructor/blob/master/core/src/storage.cpp#L74-L75 :cat2:

Reply:
Thanks. I'm surprised that changing the order back (4b4f8f0b6db7f554b3c8a95a311adf98714de36b) doesn't have any impact on any of our tests!
Shouldn't at least the pruning tests be affected?

Reply:
No, because I also [adapted the internal logic testing for `pruned()` instead of `!enabled()`](https://github.com/ros-planning/moveit_task_constructor/pull/287/commits/767d410203c60b59b4e2bce86be875ec7b0caf5b)

Reply:
Ah, sorry I guess I misunderstood. Catching the early return breaking requires a rather sophisticated test with multiple solution candidates partially getting disabled. I'm not surprised we do not have a test for that condition, though the pick-place demo could hit it - it's just hard to spot and not reliable in that context.
I encourage you to write more tests for pruning of various situations (and possibly put it in a separate test file...). :-)